ts/components/wallet/PayWebViewModal.tsx

```diff
@@ -0,0 +1,177 @@
+// the javascript submit command
+const injectedJSPostForm: string = `<script type="application/javascript">document.getElementById("${formId}").submit();</script>`;
+
+// create an html form giving a form data and an uri
+const crateAutoPostForm = (
+  form: Record<string, unknown>,
+  uri: string
+): string =>
+  `<form action="${uri}" method="post" id="${formId}" enctype="multipart/form-data" style="display: none;">
+    ${_.toPairs(form)
+      .map(kv => `<input type="text" name="${kv[0]}" value="${kv[1]}">`)
+      .join("<br/>")}
+  </form>`;
+
+/**
+ * return a 2-tuple
+ * 0 element: boolean. true is the given {@param url} contains the given {@param finishPathName} path name
+ * 0 element: string | undefined. a string contains the value of {@param outcomeQueryparamName} in query string of {@param url}
+ * @param url
+ * @param finishPathName
+ * @param outcomeQueryparamName
+ */
+const getFinishAndOutcome = (
+  url: string,
+  finishPathName: string,
+  outcomeQueryparamName: string
+): [isFinish: boolean, outComeCode: string | undefined] => {
+  const urlParse = new URLParse(url, true);
+  // find the object entry name in case insensitive
+  const maybeEntry = _.toPairs(urlParse.query).find(
+    kv => kv[0].toLowerCase() === outcomeQueryparamName.toLowerCase()
+  );
+  const outcome = maybeEntry ? maybeEntry[1] : undefined;
+  // found exit path
+  if (urlParse.pathname.toLowerCase().includes(finishPathName.toLowerCase())) {
+    return [true, outcome];
+  }
+  return [false, outcome];
+};
+
+// a loading component rendered during the webview loading times
+const renderLoading = () => (
+  <View style={styles.refreshIndicatorContainer}>
+    <RefreshIndicator />
+  </View>
+);
+
+/**
+ * A modal including a webview component.
+ * It must be used to handle a payment.
+ * A payment could start
+ * - on credit card on-boarding
+ * - on a regular payment
+ * see https://docs.google.com/document/d/1FUkW7nwHlcmN2GrBWq1YnAzoNuvcvxa7WM0_VTkYV-I/edit#heading=h.v23dcgmq0ypp
+ *
+ * this is how this component works:
+ * - at the start up, it creates a temporary html hidden form with all pre-set data (props.formFata)
+ * - it appends, at the end of that html, a JS script to auto-submit the created form to the given url (props.postUri)
+ * - it sets that html as starting page
+ * - on each page load request it checks the url
+ *    - if it is the exit url
+ *    - if it contains the outcome code
+ * - when the the exit url is found, it doesn't load it and call the handler props.onFinish passing the found (maybe not) outcome value
+ */
+export const PayWebViewModal = (props: Props) => {
+  const [outcomeCode, setOutcomeCode] = React.useState<string | undefined>(
+    undefined
+  );
+  useHardwareBackButton(() => {
+    props.onGoBack();
+    return true;
+  });
+
+  const handleOnShouldStartLoadWithRequest = (navState: WebViewNavigation) => {
+    if (navState.url) {
+      const [isFinish, maybeOutcome] = getFinishAndOutcome(
```

Comment:
Related to the comment on the `getFinishAndOutcome` function.
Consider this case:

- **finishPathName**: `foo/exit`
- **outcomeQueryparamName**: `myoutcome`

Now if inside the WebView there is this sequence of redirects:
1. `https://yourdomain.com/foo/notAFinishPath?myoutcome=123` 
    - `isFinish==false`, `maybeOutcome=123` -> outcomeCode=123
2. `https://yourdomain.com/foo/exit`
    - `isFinish==true`, `maybeOutcome=undefined` ->  outcomeCode=123

Reply:
I wrote that code respecting your scenario, I guess it is not an error. 

We have no guarantee that the `outcomeCode` is within the exit url
So that code checks for `outcomeCode` on every request. It will be returned on exit even if is found previusly